Documentation/design.md

```diff
@@ -38,6 +38,13 @@ The `endpoints` section of the `ServiceMonitorSpec`, is used to configure which
 > Note: `endpoints` (lowercase) is the field in the `ServiceMonitor` CRD, while `Endpoints` (capitalized) is the Kubernetes object kind.
 
 While `ServiceMonitor`s must live in the same namespace as the `Prometheus` resource, discovered targets may come from any namespace. This is important to allow cross-namespace monitoring use cases, e.g. for meta-monitoring. Using the `namespaceSelector` of the `ServiceMonitorSpec`, one can restrict the namespaces the `Endpoints` objects are allowed to be discovered from.
+To discover targets in all namespaces the `namespaceSelector` has to be empty:
+```yaml
+spec:
+  namespaceSelector:
```

Comment:
Have you tried this?

```yaml
spec:
  namespaceSelector:
    any: true
```

That's how it should work, if it doesn't, it's a bug :)

Reply:
Also note, that that will require very broad RBAC roles, which you may want to avoid, as in following the principle of least privilege.

Reply:
No, I didn't see that option. It's way more descriptive and I will try it! 👍

Reply:
It works. Sending an updated PR in a minute.